packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -266,27 +260,30 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 			const updateProviders = message.content.updateProviders;
 			this.refreshBroadcastRequested = true;
 			// We must be connected to receive this message, so clientId should be defined.
-			// If it isn't then, not really a problem; just won't be in provider list.
+			// If it isn't then, not really a problem; just won't be in provider or quorum list.
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-			if (updateProviders.includes(this.runtime.clientId!)) {
+			const clientId = this.runtime.clientId!;
+			if (updateProviders.includes(clientId)) {
 				// Send all current state to the new client
 				this.broadcastAllKnownState();
 			} else {
 				// Schedule a broadcast to the new client after a delay only to send if
 				// another broadcast hasn't been seen in the meantime. The delay is based
-				// on the position in the audience list. It doesn't have to be a stable
+				// on the position in the quorum list. It doesn't have to be a stable
 				// list across all clients. We need something to provide suggested order
 				// to prevent a flood of broadcasts.
-				let indexOfSelf = 0;
-				for (const clientId of this.runtime.getAudience().getMembers().keys()) {
-					if (clientId === this.runtime.clientId) {
-						break;
-					}
-					indexOfSelf += 1;
-				}
-				const waitTime = indexOfSelf * 20 + 200;
+				const quorumMembers = this.runtime.getQuorum().getMembers();
+				const indexOfSelf =
+					quorumMembers.get(clientId)?.sequenceNumber ??
+					// Index past quorum members + arbitrary additional offset up to 10
+					quorumMembers.size + Math.random() * 10;
```

Comment:
Turns out quorum member's sequenceNumber is not quite what I thought it was. It is correctly documented, but I didn't get it. These numbers will be relative to others. Can't just use the number as self index. Will need a fix.